tests/common.rs

```diff
@@ -1,482 +1,160 @@
-    // go to the Wikipedia frontpage this time
-    c.goto("https://www.wikipedia.org/").await?;
-
-    // find search input element
-    let mut e = c.wait_for_find(Locator::Id("searchInput")).await?;
-    e.send_keys("foobar").await?;
-    assert_eq!(
-        e.prop("value")
-            .await?
-            .expect("input should have value prop")
-            .as_str(),
-        "foobar"
-    );
-
-    e.clear().await?;
-    assert_eq!(
-        e.prop("value")
-            .await?
-            .expect("input should have value prop")
-            .as_str(),
-        ""
-    );
-
-    let mut c = e.client();
-    c.close().await
-}
-
-async fn raw_inner(mut c: Client) -> Result<(), error::CmdError> {
-    // go back to the frontpage
-    c.goto("https://www.wikipedia.org/").await?;
-
-    // find the source for the Wikipedia globe
-    let mut img = c.find(Locator::Css("img.central-featured-logo")).await?;
-    let src = img.attr("src").await?.expect("image should have a src");
-
-    // now build a raw HTTP client request (which also has all current cookies)
-    let raw = img.client().raw_client_for(Method::GET, &src).await?;
-
-    // we then read out the image bytes
-    let pixels = hyper::body::to_bytes(raw.into_body())
-        .map_err(error::CmdError::from)
-        .await?;
-
-    // and voilla, we now have the bytes for the Wikipedia logo!
-    assert!(pixels.len() > 0);
-    println!("Wikipedia logo is {}b", pixels.len());
-
-    c.close().await
-}
-
-async fn window_size_inner(mut c: Client) -> Result<(), error::CmdError> {
-    c.goto("https://www.wikipedia.org/").await?;
-    c.set_window_size(500, 400).await?;
-    let (width, height) = c.get_window_size().await?;
-    assert_eq!(width, 500);
-    assert_eq!(height, 400);
-
-    c.close().await
+pub fn handle_test_error(res: Result<Result<(), fantoccini::error::CmdError>, Box<dyn std::any::Any + Send>>) -> bool{
+    match res {
+        Ok(Ok(_)) => true,
+        Ok(Err(e)) => {
+            eprintln!("test future failed to resolve: {:?}", e);
+            false
+        }
+        Err(e) => {
+            if let Some(e) = e.downcast_ref::<error::CmdError>() {
+                eprintln!("test future panicked: {:?}", e);
+            } else if let Some(e) = e.downcast_ref::<error::NewSessionError>() {
+                eprintln!("test future panicked: {:?}", e);
+            } else {
+                eprintln!("test future panicked; an assertion probably failed");
+            }
+            false
+        }
+    }
 }
 
-async fn window_position_inner(mut c: Client) -> Result<(), error::CmdError> {
-    c.goto("https://www.wikipedia.org/").await?;
-    c.set_window_size(200, 100).await?;
-    c.set_window_position(0, 0).await?;
-    c.set_window_position(1, 2).await?;
-    let (x, y) = c.get_window_position().await?;
-    assert_eq!(x, 1);
-    assert_eq!(y, 2);
-
-    c.close().await
+#[macro_export]
+macro_rules! local_tester {
```

Comment:
This and `tester!` are almost identical. We can probably merge them in some reasonable way. But if you're not familiar with macros, it can be a bit of a pain, so I can do that separately later :)

Reply:
I tried earlier, but ran into problems with moving the port into an "async" closure so the test functions could be started in the same way because async closures aren't really a thing yet. I'm sure it can be done, but I'll hold off for now.